src/appshell/view/dockwindow/docktypes.h

```diff
@@ -27,8 +27,8 @@
 #include <QVariant>
 
 namespace mu::dock {
-static const char* CONTEXT_MENU_MODEL_PROPERTY("contextMenuModel");
-static const char* DOCK_PANEL_PROPERY("dockPanel");
+inline const char* CONTEXT_MENU_MODEL_PROPERTY("contextMenuModel");
+inline const char* DOCK_PANEL_PROPERY("dockPanel");
```

Comment:
making use of a recently leared C++17 feature ;-)

Reply:
In this case, it could even be `constexpr` (which implies `inline`)

Reply:
Nope: "ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]"
                                                               ^

Reply:
Or `constexpr` instead of `inline` and not instead of `nline const`?

Reply:
Yes, I meant `constexpr const char*`. I believe `constexpr` implies const for the pointer, so you need another `const` for the char it points to.

Reply:
the ìnline` looks better to me than a `constexpr const`...

Reply:
Just for fun, wouldn't inline create new copies of the string and thus use up more memory? Instead of using the same string.

Reply:
Not as var as I read the C++17 standard

Reply:
See <also https://github.com/musescore/MuseScore/pull/8897#pullrequestreview-735821234 which uses the same trick

Reply:
From [C++17 N4659 standard draft](https://github.com/cplusplus/draft/blob/master/papers/n4659.pdf), 10.1.6 "The inline specifier":
```
6 An inline function or variable with external linkage shall have the same address in all translation units.
```